### PR TITLE
feat!: simulator

### LINKED ISSUE #72: Add simulator support
Add CosmosSDK simulator support.

Currently blocked by https://github.com/strangelove-ventures/poa/issues/170 and https://github.com/strangelove-ventures/tokenfactory/issues/11

### PR DESCRIPTION
**Pull request by fmorency**
Fixes #72 

Breaking change: New Ante Handler disabling `MsgWithdrawDelegatorRewards`

I need to set the commission rate floor/ceil during application creation as the POA simulation expects `[0.1, 0.5]`. I'd like to know if there's a better approach to what I did, I don't like it very much.

To execute the full app simulation, run `make sim-app-full`

TODO
- [x] Fix determinism
- [x] Simulate Burn coins
- [x] Simulate Update params
- [ ] Write migration (maayyybe)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -883,7 +889,7 @@ func NewApp(
 	return app
 }
 
-func (app *ManifestApp) setAnteHandler(txConfig client.TxConfig) {
+func (app *ManifestApp) setAnteHandler(txConfig client.TxConfig, commissionRateMinMax RateMinMax) {
```
Comment (fmorency) [inline app/app.go:892]: The commission rate floor/ceil is set here, called by `NewApp()`. I can't set a new anteHandler after app creation as it is sealed. Happy to know if there's a less invasive way to do the same without modifying `NewApp()`'s signature.

Comment (Reecepbcups) [reply app/app.go:876]: I do not think there is a way, this is a priv method anyways so does not affect anything outside

File: app/export.go [hunk 2]
```diff
@@ -76,131 +75,15 @@ func (app *ManifestApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAdd
 
 	/* Handle fee distribution state. */
 
-	// withdraw all validator commission
-	err := app.StakingKeeper.IterateValidators(ctx, func(_ int64, val stakingtypes.ValidatorI) (stop bool) {
```
Comment (Reecepbcups) [inline app/export.go:80]: removing this logic scares me. starting a network, exporting, forking the import, and restart with same vals would need to be tested for this. Not sure of any e2e which has this

**Review by Reecepbcups — APPROVED** [review]
just that comment, if that is an acceptable risk (since it would be forking anyways) then approved

File: app/export.go (hunk 2 above)
Comment (fmorency) [reply app/export.go]: Reverted in [0ceec03](https://github.com/liftedinit/manifest-ledger/pull/76/commits/0ceec034539342d9b25fec8b97f79da8e02c1336)

**Comment by fmorency** [comment]
Ran 6 full app simulators ([a735771](https://github.com/liftedinit/manifest-ledger/pull/76/commits/a7357718e97ba770bdcf7678ed346a00b0d3bed5)), in parallel, in a loop, over the weekend. Got 0 failure.

**Pull request merged**